### ISSUE TITLE
When localStorage is not supported (e.g. with wkhtmltopdf) LoopBackAuth breaks on loading

### ISSUE THREAD

**Issue by florianheinze** [strongloop/loopback-sdk-angular#163]
My fix is to set fallbacks for localStorage and sessionStorage:

```javascript
...
  .factory('LoopBackAuth', function() {
    // fix for wkhtmltopdf
    var localStorage   = window.localStorage   || {};
    var sessionStorage = window.sessionStorage || {};
...
```

This can be done in the generated lb-services.js or in the template in lib/services.template.ejs (in line 255 at the moment) in this SDK.

The reason is that localStorage is null and when the script tries to set/get localStorage[key], wkhtmltopdf throws an "Warning: undefined:0 TypeError: 'null' is not an object" error.

**Comment by davidcheung** [comment]
Hi @florianheinze, Thanks for raising the issue, Looks like it is related to #179, and likely will be fixed upon handling `localStorage` fallback for Safari as well. Tagging the issue for future reference to handle scenarios such as `wkhtmltopdf `.

**Comment by davidcheung** [comment]
Related : https://github.com/strongloop/loopback-sdk-angular/pull/203

**Comment by davidcheung** [comment]
Hi @florianheinze, would you be able to verify if this issue is fixed by #203?
Thanks!

**Comment by davidcheung** [comment]
Hi @florianheinze, 
Closing this issue, if ` var localStorage   = window.localStorage   || {};` was able to solve that issue, I believe the try catch should also be able to handle this as it should be a caught error now, please reopen if issue persists. Thanks!
